init/fluent_bit_init_process.go

```diff
@@ -0,0 +1,401 @@
+	defer fileFromS3.Close()
+
+	_, err := s3Downloader.Download(fileFromS3,
+		&s3.GetObjectInput{
+			Bucket: aws.String(bucketName),
+			Key:    aws.String(s3FilePath),
+		})
+	if err != nil {
+		logrus.Errorln(err)
+		logrus.Fatalf("[FluentBit Init Process] Cannot download %s from s3\n", s3FileName)
+	}
+}
+
+func processS3ConfigFiles(folderPath string) {
+
+	fileInfos, err := ioutil.ReadDir(folderPath)
+	if err != nil {
+		logrus.Errorln(err)
+		logrus.Fatalf("[FluentBit Init Process] Unable to read config files in %s folder\n", folderPath)
+	}
+
+	for _, file := range fileInfos {
+		filePath := folderPath + file.Name()
+		processConfigFile(filePath)
+	}
+}
+
+// use @INCLUDE to add config files to the main config file
+func writeInclude(configFilePath, mainConfigFilePath string) {
+
+	mainConfigFile := openFile(mainConfigFilePath)
+	defer mainConfigFile.Close()
+
+	writeContent := "@INCLUDE " + configFilePath + "\n"
+	_, err := mainConfigFile.WriteString(writeContent)
+	if err != nil {
+		logrus.Errorln(err)
+		logrus.Fatalf("[FluentBit Init Process] Cannot write %s in main config file: %s\n", writeContent[:len(writeContent)-2], mainConfigFilePath)
+	}
+}
+
+// change the fluent bit cammand to use "-R" to specift Parser config file
+func updateCommand(parserFilePath string) {
+	fluentBitCommand = fluentBitCommand + " -R " + parserFilePath
+	logrus.Infoln("[FluentBit Init Process] Command is change to -> " + fluentBitCommand)
+}
+
+// change the invoke_fluent_bit.sh
+// which will declare ECS Task Metadata as environment variables
+// and finally invoke Fluent Bit
+func modifyInvokeFile(filePath string) {
+
+	invokeFile := openFile(filePath)
+	defer invokeFile.Close()
+
+	_, err := invokeFile.WriteString(fluentBitCommand)
+	if err != nil {
+		logrus.Errorln(err)
+		logrus.Fatalf("[FluentBit Init Process] Cannot write %s in invoke_fluent_bit.sh\n", fluentBitCommand)
+	}
+}
+
+// create a file, when flag is true, the file will be closed automatically after creation
+func createFile(filePath string, AutoClose bool) *os.File {
+
+	file, err := os.Create(filePath)
+	if err != nil {
+		logrus.Errorln(err)
+		logrus.Fatalf("[FluentBit Init Process] Cannot create the file: %s\n", filePath)
+	}
+
+	if AutoClose {
+		defer file.Close()
+	}
+
+	return file
+}
+
+func openFile(filePath string) *os.File {
+
+	file, err := os.OpenFile(filePath, os.O_APPEND|os.O_WRONLY, 0700)
+	if err != nil {
+		logrus.Errorln(err)
+		logrus.Fatalf("[FluentBit Init Process] Unable to read %s\n", filePath)
+	}
+	return file
+}
+
+func main() {
+
+	// create the invoke_fluent_bit.sh
+	// which will declare ECS Task Metadata as environment variables
+	// and finally invoke Fluent Bit
+	createFile(invokeFile, true)
+
+	// get ECS Task Metadata and set the region for S3 client
+	httpClient := createHTTPClient()
+	metadata := getECSTaskMetadata(httpClient)
+	metadataReigon = reflect.ValueOf(metadata).Field(0).Interface().(string)
+
```

Comment:
I like the format in `main()`, it's clean and in good shape. 
These three lines could be optimized to only one function like others in `main()` as well.

and move these three lines into that function outside of `main()`.

Reply:
Writing them out for unit testing, `metadataReigon` has removed a from the `main()`. 
But I think I still have to keep the other two..